macros/tests.sql

```diff
@@ -78,37 +78,58 @@
   {% set expectations = test_configuration.expectations %}
   {% set model_node = dbt_unit_testing.model_node(test_configuration.model_node) %}
   {%- set model_complete_sql = dbt_unit_testing.build_model_complete_sql(model_node, test_configuration.mocks, test_configuration.options) -%}
-  {% set columns = dbt_unit_testing.quote_and_join_columns(dbt_unit_testing.extract_columns_list(expectations.input_values)) %}
+
+  {% set all_columns = dbt_unit_testing.extract_columns_list(expectations.input_values) %}
+  {% set convert_columns = test_configuration.options.convert_columns | default("convert_columns") %}
+  {% set selected_columns = dbt_unit_testing.quote_and_join_columns(all_columns, convert_columns) %}
+  {% set grouped_columns = dbt_unit_testing.quote_and_join_columns(all_columns) %}
 
   {% set diff_column = test_configuration.options.diff_column | default("diff") %}
   {% set count_column = test_configuration.options.count_column | default("count") %}
 
   {%- set actual_query -%}
-    select count(1) as {{ count_column }}, {{columns}} from ( {{ model_complete_sql }} ) as s group by {{ columns }}
+    actual as (
+    with
```

Comment:
I don't think embedding ctes like this would work

Reply:
This code works on BigQuery. What adapters is the package intended to support? Since profiles.yml is not included in the project, it's hard for me to determine. I usually extract all the secrets and commit profiles.yml in the repo.

dbt labs uses the dispatch_adapter method to customise queries for different sql engines but the code in this package does not currently implement that pattern.